eng/ci/templates/official/jobs/run-benchmarks.yml

```diff
@@ -91,12 +92,55 @@ jobs:
         Write-Host "Running command: $command"
         Invoke-Expression $command
 
+  # Retrieve function host logs
   - task: PowerShell@2
-    displayName: Functions host logs
+    displayName: 'Fetch Function Host Logs'
     inputs:
       targetType: 'inline'
       script: |
         $url = "$(crankAgentUrl)/jobs/1/output"
-        Write-Host "Making GET request to: $url to get logs"
+        Write-Host "Fetching logs from: $url"
         $response = Invoke-WebRequest -Uri $url -Method GET -UseBasicParsing
         Write-Host $response.Content
+
+  # Tag the build as a baseline if it originates from the specified branch.
+  # Baseline builds serve as reference points for performance comparisons in future builds.
+  # The tag added here will help identify these builds in the pipeline.
+  - task: PowerShell@2
+    condition: and(succeeded(), eq(variables['sourceBranch'], variables['benchmarkBaselineBranch']))
+    displayName: 'Tag Build as Baseline'
+    inputs:
+      targetType: 'inline'
+      script: Write-Host "##vso[build.addbuildtag]$(benchmarkBaselineTagName)"
+
+  # Download baseline result
+  - download: BaselineResult
```

Comment:
I believe `download` task is blocked / non-compliant in 1ES and you need to use the templateContext `inputs` section.

Reply:
Switched to `templateContext.inputs` approach. (As per the docs, both are valid, but the docs recommend using templateContext.inputs`)